Added GMock to UT project compilation

mbronk opened the pull request:
Rationale: Allows to use full portfolio of matchers/assertions.
E.g. http://google.github.io/googletest/reference/matchers.html#container-matchers

Codebase is included with GTest (=was already there, just not compiled)

mbronk committed:
Added GMock to UT Makefile

Signed-off-by: Mateusz Bronk bronk.m+gh@gmail.com

mbronk commented on test/ir_Argo_test.cpp:
```diff
@@ -50,9 +51,11 @@ TEST(TestArgoACClass, MessageConstructon) {
+  auto expected = std::vector<uint8_t>({
+      0xAC, 0xF5, 0x00, 0x24, 0x02, 0x00, 0x00, 0x00, 0x00, 0xAC, 0xD6, 0x01});
+  auto actual = ac.getRaw();
+  EXPECT_THAT(std::vector<uint8_t>(actual, actual + kArgoBits / 8),
```
This is mostly to prove out that CI links w/ GMock correctly.
The benefit in using `::ElementsAreArray` over `EXPECT_STATE_EQ` is **marginal** in this case (the only material difference is GMock's matcher would spit out all differences, and not stop on first error).

----
The wrapping into vector is there, mostly to include readability (I don't believe the extra copy for UT hurts anything).
In theory this could be written like so:
```
  EXPECT_THAT(*reinterpret_cast<uint8_t(*)[kArgoBits / 8]>(ac.getRaw()),
              ::testing::ElementsAreArray(expected));
````

Anyways, this is mostly a sample to prove integration works. As long as the i-face is a C-Style array, use of fancy matchers doesn't bring much benefit (but they start to shine once STL containers get used)

NiKiZe commented on test/Makefile:
```diff
@@ -24,11 +25,14 @@ INCLUDES = -I$(USER_DIR) -I.
 # Set Google Test's header directory as a system directory, such that
 # the compiler doesn't generate warnings in Google Test headers.
-CPPFLAGS += -isystem $(GTEST_DIR)/include -DUNIT_TEST -D_IR_LOCALE_=en-AU
+CPPFLAGS += -isystem $(GTEST_DIR)/include  -isystem $(GMOCK_DIR)/include -DUNIT_TEST -D_IR_LOCALE_=en-AU
```
Not, double space?

mbronk replied:
Right, fixed

NiKiZe commented on test/Makefile:
```diff
@@ -92,24 +104,39 @@ COMMON_DEPS = $(USER_DIR)/IRrecv.h $(USER_DIR)/IRsend.h $(USER_DIR)/IRtimer.h \
         $(GTEST_DIR)/src/gtest_main.cc
 
+gmock-all.o : $(GMOCK_SRCS_)
+	$(CXX) $(CPPFLAGS) -I$(GTEST_DIR) -I$(GMOCK_DIR) $(CXXFLAGS) \
```
I think it is possible to use wildcard rules here?

mbronk replied:
Absolutely, though these "GMock" Makefile receipes are fairly generic and written this way. 
See https://github.com/ceph/gmock/blob/master/make/Makefile for example

mbronk commented:
Hi,
This PR is really to test the waters :) 
I'm currently adding support for new type of Argo AC remote and figured I could use some of GMock's goodness in the UTs I'm adding there _(I personally believe GTest+GMock are much more powerful when used together)_.

Since it was just lying there... I added `GMock` to compilation (at the moment it is not doing anything useful, the only "Argo" file I changed was to prove out CI can compile and link with it w/o issues). 
I can also revert that sample change if you like (though IMHO it is adding 0.01% of value still).

@crankyoldgit - are you OK with this direction?

crankyoldgit commented:
> Hi, This PR is really to test the waters :) I'm currently adding support for new type of Argo AC remote and figured I could use some of GMock's goodness in the UTs I'm adding there _(I personally believe GTest+GMock are much more powerful when used together)_.
> 
> Since it was just lying there... I added `GMock` to compilation (at the moment it is not doing anything useful, the only "Argo" file I changed was to prove out CI can compile and link with it w/o issues). I can also revert that sample change if you like (though IMHO it is adding 0.01% of value still).
> 
> @crankyoldgit - are you OK with this direction?

Yep. I'm fine with it.

crankyoldgit reviewed (approved):
LGTM!
